tests/testerina-integration-test/src/test/java/org/ballerinalang/testerina/test/TestReportTest.java

```diff
@@ -76,70 +81,111 @@ public void testWarningForJacocoXMLFlag() throws BallerinaTestException {
+
+        int[] mathStatus =  {mathTotal, mathPassed, mathFailed, mathSkipped};
+        int[] fooStatus =  {fooTotal, fooPassed, fooFailed, fooSkipped};
+        int[] bartestStatus =  {bartestTotal, bartestPassed, bartestFailed, bartestSkipped};
+
+        validateStatuses(mathStatus, fooStatus, bartestStatus);
         validateCoverage();
     }
 
-    @Test (enabled = false)
-    public void testWithoutCoverage() throws BallerinaTestException, IOException {
-        runCommand(false);
-        validateStatuses();
+    @Test(enabled = false)
+    public void testWithoutCoverage() throws BallerinaTestException {
+        String[] args = new String[]{"--test-report"};
+        runCommand(args);
+
+        int mathTotal = 2, mathPassed = 2, mathFailed = 0, mathSkipped = 0;
+        int fooTotal = 2, fooPassed = 0, fooFailed = 1, fooSkipped = 1;
+        int bartestTotal = 1, bartestPassed = 1, bartestFailed = 0, bartestSkipped = 0;
+
+        int[] mathStatus =  {mathTotal, mathPassed, mathFailed, mathSkipped};
+        int[] fooStatus =  {fooTotal, fooPassed, fooFailed, fooSkipped};
+        int[] bartestStatus =  {bartestTotal, bartestPassed, bartestFailed, bartestSkipped};
+
+        validateStatuses(mathStatus, fooStatus, bartestStatus);
         Assert.assertEquals(resultObj.get("moduleCoverage").getAsJsonArray().size(), 0);
     }
 
-    private void runCommand(boolean coverage) throws BallerinaTestException, IOException {
-        String[] args;
-        if (coverage) {
-            args = new String[]{"--code-coverage", "--includes=*"};
-        } else {
-            args = new String[]{"--test-report"};
-        }
-        balClient.runMain("test", args, null, new String[]{},
-                new LogLeecher[]{}, projectPath);
+    @Test()
+    public void testModuleWiseCoverage() throws BallerinaTestException {
+        String[] args = new String[]{"--code-coverage", "--tests", "foo.math:*"};
+        runCommand(args);
+
+        int mathTotal = 2, mathPassed = 2, mathFailed = 0, mathSkipped = 0;
+        int fooTotal = 0, fooPassed = 0, fooFailed = 0, fooSkipped = 0;
+        int bartestTotal = 0, bartestPassed = 0, bartestFailed = 0, bartestSkipped = 0;
+
+        int[] mathStatus =  {mathTotal, mathPassed, mathFailed, mathSkipped};
+        int[] fooStatus =  {fooTotal, fooPassed, fooFailed, fooSkipped};
+        int[] bartestStatus =  {bartestTotal, bartestPassed, bartestFailed, bartestSkipped};
+
+        validateStatuses(mathStatus, fooStatus, bartestStatus);
+        validateModuleWiseCoverage();
+    }
+
+    private void runCommand(String[] args) throws BallerinaTestException {
+        balClient.runMain("test", args, null, new String[]{}, new LogLeecher[]{}, projectPath);
+
         Gson gson = new Gson();
-        BufferedReader bufferedReader = Files.newBufferedReader(resultsJsonPath, StandardCharsets.UTF_8);
-        resultObj = gson.fromJson(bufferedReader, JsonObject.class);
+
+        try {
```

Comment:
Let's use try with resources here.

`try(BufferedReader bufferedReader = Files.newBufferedReader(resultsJsonPath, StandardCharsets.UTF_8)){`

Reply:
Made use of `try-with-resources` for the buffered reader